src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.ts

```diff
@@ -0,0 +1,74 @@
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+import _ from 'lodash';
+
+import { Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+
+@Directive({
+  selector: '[cdScope]'
+})
+export class AuthStorageDirective {
+  permissions: Permissions;
+  matchAll = false;
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private authStorageService: AuthStorageService
+  ) {}
+
+  @Input('cdScope') set cdScope(condition: string | string[] | object) {
+    this.permissions = this.authStorageService.getPermissions();
+    if (this.isAuthorized(condition)) {
+      this.viewContainer.createEmbeddedView(this.templateRef);
+    } else {
+      this.viewContainer.clear();
+    }
+  }
+
+  @Input('cdScopeMatchAll') set cdScopeMatchAll(condition: boolean) {
+    this.matchAll = condition;
+  }
```

Comment:
Just for setting a variable, you doesn't really need a setter (that's needed when you want to trigger side-effects when that variable changes, as in the case of the `cdScope`, where you hide/display the container):
```suggestion
  @Input() cdScopeMatchAll: boolean = False;
```

Reply:
Nice! a oneliner

Reply:
@epuertat please I don't understand how to use this

Reply:
Its ok, I got it